au4/src/projectscene/qml/Audacity/ProjectScene/trackspanel/audio/VolumePressureMeter.qml

```diff
@@ -60,43 +67,117 @@ Canvas {
         property bool rulerNeedsPaint: true
     }
 
-    function drawRuler(ctx, originVPos, originHPos, fullStep, smallStep, strokeHeight, strokeWidth) {
-        ctx.clearRect(0, prv.indicatorWidth, root.height, root.width - prv.indicatorWidth)
-        ctx.font = prv.unitTextFont
-
-        var currentStrokeVPos = 0
-
-        for (var i = 0; i <= prv.fullValueRangeLength; i+=smallStep) {
-            if (i == 0) {
-                currentStrokeVPos = originVPos
-            } else {
-                currentStrokeVPos += prv.divisionPixels * smallStep
+    // Rounding up fullStep value to the predefined one,
+    // to avoid getting funny intervals like 3, or 7
+    function roundUpToFixedValue(value) {
+        // full and small step is
+        // a number units per each respective notch on the ruler
+        const steps = [
+            { fullStep: 1, smallStep: 0 },
+            { fullStep: 2, smallStep: 1 },
+            { fullStep: 5, smallStep: 1 },
+            { fullStep: 10, smallStep: 2 },
+            { fullStep: 20, smallStep: 4 },
+            { fullStep: 50, smallStep: 10 },
+            { fullStep: 100, smallStep: 20 }
+        ];
+
+        // Find the nearest full step
+        for (let i = 0; i < steps.length; i++) {
+            if (value <= steps[i].fullStep) {
+                return steps[i];
             }
+        }
 
-            if (i % fullStep) {
-                ctx.fillStyle = prv.shortStrokeColor
-                ctx.fillRect(currentStrokeVPos,
-                             originHPos,
-                             prv.shortStrokeHeight,
-                             prv.shortStrokeWidth)
+        // Should not happen
+        return steps[steps.length - 1];
+    }
 
-            } else {
-                ctx.fillStyle = prv.longStrokeColor
-                ctx.fillRect(currentStrokeVPos,
-                             originHPos,
-                             prv.longStrokeHeight,
-                             prv.longStrokeWidth)
+    // Draws a rectangle rounded at the top, bottom or both
+    function drawRoundedRect(ctx, fillStyle, x, y, width, height, radius, roundedEdge) {
```

Comment:
Sounds like a function we'd like to re-use somewhere probably, helper qml module would be nice maybe? But that's just something to think of, not necessarily now

Reply:
I would do that if you could guide me on where to put it

Reply:
Usually we have an approach that if it is used in more than two places, only then we move it to a separate place, so we need to wait
separate place - we have `UiComponents/Utils.js` for such tasks